doc/under_sampling.rst

```diff
@@ -125,75 +125,96 @@ It would also work with pandas dataframe::
+will explain in the coming paragraphs. We can perform this undersampling as follows::
 
   >>> from imblearn.under_sampling import NearMiss
   >>> nm1 = NearMiss(version=1)
   >>> X_resampled_nm1, y_resampled = nm1.fit_resample(X, y)
   >>> print(sorted(Counter(y_resampled).items()))
   [(0, 64), (1, 64), (2, 64)]
 
-As later stated in the next section, :class:`NearMiss` heuristic rules are
-based on nearest neighbors algorithm. Therefore, the parameters ``n_neighbors``
-and ``n_neighbors_ver3`` accept classifier derived from ``KNeighborsMixin``
-from scikit-learn. The former parameter is used to compute the average distance
-to the neighbors while the latter is used for the pre-selection of the samples
-of interest.
 
 Mathematical formulation
-^^^^^^^^^^^^^^^^^^^^^^^^
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+:class:`NearMiss` uses the K-Nearest Neighbour algorithm to identify the samples of the
+target class(es) that are closer to the minority class, as well as the distance that
+separates them.
 
-Let *positive samples* be the samples belonging to the targeted class to be
-under-sampled. *Negative sample* refers to the samples from the minority class
-(i.e., the most under-represented class).
+Let *positive samples* be the samples belonging to the class to be under-sampled, and
+*negative sample* the samples from the minority class (i.e., the most
+under-represented class).
 
-NearMiss-1 selects the positive samples for which the average distance
-to the :math:`N` closest samples of the negative class is the smallest.
+**NearMiss-1** selects the positive samples whose average distance to the :math:`K`
+closest samples of the negative class is the smallest (:math:`K` is the number of
+neighbours in the K-Nearest Neighbour algorithm). The following image illustrates the
+logic:
 
 .. image:: ./auto_examples/under-sampling/images/sphx_glr_plot_illustration_nearmiss_001.png
    :target: ./auto_examples/under-sampling/plot_illustration_nearmiss.html
    :scale: 60
    :align: center
 
-NearMiss-2 selects the positive samples for which the average distance to the
-:math:`N` farthest samples of the negative class is the smallest.
+**NearMiss-2** selects the positive samples whose average distance to the
+:math:`K` farthest samples of the negative class is the smallest. The following image
+illustrates the logic:
 
 .. image:: ./auto_examples/under-sampling/images/sphx_glr_plot_illustration_nearmiss_002.png
    :target: ./auto_examples/under-sampling/plot_illustration_nearmiss.html
    :scale: 60
    :align: center
 
-NearMiss-3 is a 2-steps algorithm. First, for each negative sample, their
-:math:`M` nearest-neighbors will be kept. Then, the positive samples selected
-are the one for which the average distance to the :math:`N` nearest-neighbors
-is the largest.
+**NearMiss-3** is a 2-steps algorithm:
+
+First, for each negative sample, that is, for each observation of the minority class,
```

Comment:
I refer back to target class and minority because positive and negative are a bit confusing, and at this stage, the reader may have forgotten which is which.